seismicpro/survey/metrics.py

```diff
@@ -29,3 +72,862 @@ def plot(self, ax, coords, index, sort_by=None, **kwargs):
+        direction = "below" if self.shift > 0 else "above"
+        msg = f"Traces with RMS {self._get_threshold_description()} computed in a {self.window_size} ms window around"\
+              f" RV shifted {direction} by {abs(self.shift)} ms "
+        return msg + self._get_threshold_description()
+
+    def describe(self, metric_values, line_width=60):
+        """Provide a description about the number of bad values for the passed metric values in a string format. Each
+        line in the resulting string will not exceed `line_width`. If `self.threshold` is None, the average RMS for
+        the passed values will be showed."""
+        metric_value = np.sqrt(metric_values[:, 0] / metric_values[:, 1])
+        if self.threshold is None:
+            direction =  "below" if self.shift > 0 else "above"
+            description = f"Mean of traces RMS computed in a {self.window_size} ms window \n"
+            description += f"{f'around RV shifted {direction} by {abs(self.shift)} ms:':<{line_width}}"
+            return description + f"{np.nanmean(metric_value):<.3f}"
+        return super().describe(metric_value, line_width=line_width)
+
+    def get_values(self, gather):
+        fbp_times = self.refractor_velocity(gather.offsets)
+        kwargs = {"traces": gather.data, "window_size": self.window_size, "shift": self.shift,
+                  "samples": gather.samples, "delay": gather.delay, "sample_interval": gather.sample_interval,
+                  "fbp_times": fbp_times, "_get_indices": self._get_indices,
+                  "compute_stats_by_ixs": self.compute_stats_by_ixs}
+        return self.numba_get_values(**kwargs)
+
+    @staticmethod
+    @njit(nogil=True)
+    # pylint: disable-next=redefined-outer-name
+    def numba_get_values(traces, window_size, shift, samples, delay, sample_interval, fbp_times, _get_indices,
+                         compute_stats_by_ixs):
+        start_ixs, end_ixs = _get_indices(window_size=window_size, shift=shift, samples=samples, delay=delay,
+                                          sample_interval=sample_interval, fbp_times=fbp_times)
+        return compute_stats_by_ixs(traces, start_ixs=start_ixs, end_ixs=end_ixs)
+
+    @staticmethod
+    @njit(nogil=True)
+    # pylint: disable-next=redefined-outer-name
+    def _get_indices(window_size, shift, samples, delay, sample_interval, fbp_times):
+        """Calculates the start and end indices of a window of size `window_size` centered around the refractor
+        velocity shifted by `shift`."""
+        mid_times = fbp_times + shift
+        start_times = np.clip(mid_times - (window_size - window_size // 2), 0, samples[-1])
+        end_times = np.clip(mid_times + (window_size // 2), 0, samples[-1])
+
+        start_ixs = np.rint((start_times - delay) / sample_interval).astype(np.int32)
+        end_ixs = np.rint((end_times - delay) / sample_interval).astype(np.int32)
+        return start_ixs, end_ixs
```

Comment:
```suggestion
        def _get_indices(fbp_times, shift, window_size, n_samples, sample_interval, delay):
        """Calculates the start and end indices of a window of size `window_size` centered around the refractor
        velocity shifted by `shift`."""
        mid_times = fbp_times + shift

        start_times = mid_times - window_size / 2
        start_ixs = (start_times - delay) / sample_interval
        start_ixs = np.clip(np.rint(start_ixs), 0, n_samples - 1).astype(np.int32)
        
        end_times = mid_times + window_size / 2
        end_ixs = (end_times - delay) / sample_interval
        end_ixs = np.clip(np.rint(end_ixs), 0, n_samples - 1).astype(np.int32)
        return start_ixs, end_ixs
```
Currently you don't handle a case of positive `delay`

Reply:
added with slightly different window_size processing.